Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/ServerHandler.java

```diff
@@ -0,0 +1,124 @@
+package com.example.android.camera2.slowmo;
+
+import android.content.ContentValues;
+import android.os.AsyncTask;
+import android.util.Log;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutionException;
+import org.jetbrains.annotations.NotNull;
+
+/** Handles every interaction of app with the server. */
+public class ServerHandler {
+
+  private Integer serverSocketPort;
+  private long serverStartTimestamp;
+  private Socket socket;
+  private BufferedReader inputReader;
+  private PrintWriter outputWriter;
+  private long hostSyncTimestamp;
+
+  public ServerHandler(Integer serverSocketPort) {
+    this.serverSocketPort = serverSocketPort;
+  }
+
+  public long getHostSyncTimestamp() {
+    return hostSyncTimestamp;
+  }
+
+  public Integer getServerSocketPort() {
+    return serverSocketPort;
+  }
+
+  public void setServerSocketPort(Integer serverSocketPort) {
+    this.serverSocketPort = serverSocketPort;
+  }
+
+  public long getServerStartTimestamp() {
+    return serverStartTimestamp;
+  }
+
+  public void startConnection() {
+    ConnectionThread connectionThread = new ConnectionThread();
+    connectionThread.execute();
+  }
+
+  public void sendKeySimulationSignal() {
+    hostSyncTimestamp = System.currentTimeMillis();
```

Comment:
leave a todo for a more robust time sync measurement (we need at least a few round trips averaged)

Reply:
done